.github/workflows/on_pr_test_python_31x.yml

```diff
@@ -6,6 +6,7 @@ on:
   pull_request:
   push:
     branches:
+      - 'macos12_ci'
```

Comment:
This PR itself should still run CI without needing to list it in the list of push branches, right?

```suggestion
```